src/util/tryCatch.ts

```diff
@@ -11,7 +11,7 @@ function tryCatcher(): any {
   }
 }
 
-export function tryCatch(fn: Function): Function {
+export function tryCatch<T extends Function>(fn: T): T {
   tryCatchTarget = fn;
-  return tryCatcher;
-};
\ No newline at end of file
+  return <any>tryCatcher;
```

Comment:
no the cast to `any` just means we're telling the compiler assume this value is whatever, then the return type will be whatever the function is defined as, this case it's `T`